watcher/watcher.go

```diff
@@ -0,0 +1,91 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/etcd/clientv3/namespace"
+
+	etcd "github.com/coreos/etcd/clientv3"
+)
+
+// Watch watches etcd directory and emits events when key/value pair was added, deleted or changed.
+// Watch function also emits events for pre-existing key/value pairs.
+func Watch(client *etcd.Client, prefix string, stopCh <-chan struct{}) (<-chan *Event, error) {
+	watcherClient := namespace.NewWatcher(client.Watcher, prefix)
+	kvClient := namespace.NewKV(client.KV, prefix)
+
+	eventsCh := make(chan *Event)
+
+	existingValues, err := list(kvClient)
+	if err != nil {
+		return nil, fmt.Errorf("listing existing values faield: %s", err)
+	}
+
+	go func() {
+		defer watcherClient.Close()
+		defer close(eventsCh)
+
+		// populate channel with events about existing values
+		for _, existingValue := range existingValues {
+			eventsCh <- existingValue
+		}
+
+		watchCh := watcherClient.Watch(context.TODO(), "", etcd.WithPrefix())
+		for resp := range watchCh {
+			select {
+			case <-stopCh:
+				return
+			default:
```

Comment:
Is this needed?

Reply:
Without that it will stuck on `case <-stopCh`